packages/botonic-plugin-flow-builder/src/content-fields/flow-button.tsx

```diff
@@ -27,14 +27,16 @@ export class FlowButton extends ContentFieldsBase {
     const newButton = new FlowButton(cmsButton.id)
     newButton.text = this.getTextByLocale(locale, cmsButton.text)
     if (cmsButton.target) {
-      newButton.payload = cmsButton.target.id
+      newButton.payload = cmsApi.getPayload(cmsButton.target)
```

Comment:
Observation: Comming back to the suggestion of creating a class for the target objects, this could be called this way:
```suggestion
      newButton.payload = cmsButton.target.getPayload()
```
or even this way if you create a `get` method.
```suggestion
      newButton.payload = cmsButton.target.payload
```